Update repository with new add-on version

AlexanderEpolite opened the pull request:
This is the big PR that was delayed for several months.

Closes #7 (for obvious reasons).
Closes #6 as the link is no longer in the README.
Closes #4
Closes #3

Not sure about issue number 2, let me know about that one.  The README might need some updating, so feel free to suggest changes.

AlexanderEpolite committed:
Replace repository with new extension

AlexanderEpolite committed:
update readme

AlexanderEpolite committed:
fix typo

AlexanderEpolite committed:
Add NodeJS to build tools

AlexanderEpolite committed:
update and rename license

AlexanderEpolite commented:
Additionally, the URL in the repository will need to be changed: https://addons.mozilla.org/en-US/firefox/addon/justdeleteme/

tupaschoal commented on README.md:
```diff
@@ -1,33 +1,32 @@
 
-This extension will add a traffic light icon to your urlbar indicating the difficulty in removing an account on the website you're visiting is.
+## Download
+You can download the extension [![at this link](https://ffp4g1ylyit3jdyti1hqcvtb-wpengine.netdna-ssl.com/addons/files/2015/11/get-the-addon.png)](https://addons.mozilla.org/en-US/firefox/addon/justdeleteme/).
```
I tried viewing the readme as `md` and seems like the image is broken.

tupaschoal commented on js/background.js:
```diff
@@ -0,0 +1,99 @@
+
+const data_url = "https://raw.githubusercontent.com/jdm-contrib/jdm/master/_data/sites.json";
+
+//taken from https://github.com/jdm-contrib/justdelete.me-firefox-addon/blob/master/lib/main.js
```
I think we can get away with this comment, since we intend to merge :P

tupaschoal commented on js/background.js:
```diff
@@ -0,0 +1,99 @@
+    /**
+     * Set the icon.
+     *
+     * @param icon {"easy" | "medium" | "hard" | "impossible" | "unknown"}
```
```suggestion
     * @param icon {"easy" | "medium" | "hard" | "impossible" | "limited" | "unknown"}
```

tupaschoal commented on js/background.js:
```diff
@@ -0,0 +1,99 @@
+     * Get the hostname from a URL.
+     * 
+     * Parodied from https://github.com/jdm-contrib/justdelete.me-firefox-addon/blob/master/lib/main.js
+     * 
```
```suggestion
```

outdated comment

tupaschoal commented on js/background.js:
```diff
@@ -0,0 +1,99 @@
+     * Search for the site and return the difficulty
+     * 
+     * @param domain {string}
+     * @returns {"easy" | "medium" | "hard" | "impossible" | undefined} the difficulty, or undefined if there is none.
```
```suggestion
     * @returns {"easy" | "medium" | "hard" | "impossible" | "limited" | undefined} the difficulty, or undefined if there is none.
```

tupaschoal commented on pub/index.html:
```diff
@@ -0,0 +1,36 @@
+</div>
+<div class="unofficial_warning">
+    <h6>Unofficial JustDeleteMe.xyz browser extension.</h6>
+</div>
```
```suggestion
```

We won't be needing that :)

tupaschoal reviewed (commented):
Thanks for coming back! I have just a few simple notes.

AlexanderEpolite committed:
Fix some documentation in the main js file.
I am on my laptop currently and do not have my private key, so this
commit is unsigned.

AlexanderEpolite committed:
add firefox addon image; HTML within notes will now display in the extension, rather than the HTML code.

AlexanderEpolite committed:
update readme to include add-on image

AlexanderEpolite committed:
done

AlexanderEpolite committed:
I erased the wrong header

AlexanderEpolite commented:
@tupaschoal The commits are unsigned because I am on my laptop right now and do not have access to my private key.  I have resolved the notes you made, let me know if there is anything else that needs to be changed before the merge.

tupaschoal commented:
Everything here looks fine @alexanderepolite , thanks!

I've invited you to the organization to have access to the extension repos. I think you might need to do some updating on the add-on webpage itself to point to its new home.

AlexanderEpolite commented:
Everything looks good!  Can you change the repository URL to https://addons.mozilla.org/en-US/firefox/addon/justdeleteme/ @tupaschoal ?  I do not have permission to change this.

tupaschoal commented:
Done!
